src/providers/auth.ts

```diff
@@ -0,0 +1,59 @@
+import {Provider, Inject} from 'angular2/core';
+import {ReplaySubject} from 'rxjs/subject/ReplaySubject';
+import {FirebaseRef} from '../tokens';
+
+import * as Firebase from 'firebase';
+
+const kBufferSize = 1; 
+
+export enum AuthProviders {
+  Github
+};
+
+export class Auth<T extends ProviderMetadata> extends ReplaySubject<AuthState<T>> {
```

Comment:
I think right now we should move away from making this generic, to make it simpler for users to use, and to make it easier to adapt this to some changes we're planning to make (like attaching this to a global AngularFire service: `af.auth.subscribe()`.

Instead of generic, could you make the `AuthState` interface have optional fields for the providers? Github is fine for now:

```ts
interface AuthState {
  github?: GithubAuthMetadata
}
```